ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Impl.hs

```diff
@@ -458,120 +463,157 @@ reOpenFile :: forall m h blockId
-                        _  -> let (so,(bs,_)) = last ls in so + bs
-                let fileMp = Map.fromList ls
-                case mErr of
-                    Nothing -> return ()
-                    Just _err -> do
-                        -- the handle of the parser is closed at this point. We need to
-                        -- reOpen the file in AppendMode now (parser opens with ReadMode).
-                        -- Note that no file is open at this point, so we can safely open
-                        -- with AppendMode any file, without the fear of opening multiple
-                        -- concurrent writers, which is not allowed.
-
-                        -- TODO(kde) we should add a Warning log here.
-                        withFile hasFS file (AppendMode AllowExisting) $ \hndl ->
-                            hTruncate hndl (fromIntegral offset)
-                        return ()
-                newRevMp <- fromEither err $ reverseMap file revMp fileMp
-                let (fileInfo, maxSlotOfFile) = FileInfo.fromParsedInfo ls
-                let newMp = Map.insert file fileInfo mp
-                let newMaxSlot = maxSlotList $ catMaybes [maxSlot, maxSlotOfFile]
-                let newSuccMp = foldr
-                        (\(_,(_, blockInfo)) pm -> insertMapSet pm (bbid blockInfo, bpreBid blockInfo))
-                        succMp
-                        ls
-                -- error here is reasonable because we have already checked that all filenames parse.
-                let fd = fromMaybe (error $ "file name " <> show file <> " failed to parse") (parseFd file)
-                let newHaveLessThanN = if FileInfo.isFull n fileInfo
-                        then haveLessThanN
-                        else (fd, file, offset) : haveLessThanN
-                go newMp newRevMp newSuccMp newMaxSlot newHaveLessThanN restFiles
-    go Map.empty Map.empty Map.empty Nothing [] (Set.toList files)
-
--- This is safe in terms of throwing FsErrors.
+-- | Makes the 'InternalState' by parsing all files.
+--
+-- This function may create a new file to append new blocks or use an existing
+-- one.
+mkInternalState
+  :: forall blockId m h e. (HasCallStack, MonadCatch m, Ord blockId)
+  => HasFS m h
+  -> ErrorHandling (VolatileDBError blockId) m
+  -> Parser e m blockId
+  -> Int
+  -> [(FileId, FsPath)]
+  -> m (InternalState blockId h)
+mkInternalState hasFS err parser n files =
+    wrapFsError (hasFsErr hasFS) err $
+      go Index.empty Map.empty Map.empty Nothing [] files
+  where
+    lastFile = safeMaximumOn fst files
+    newFileInfo mp newIndex =
+      ( filePath newIndex
+      , newIndex
+      , newIndex + 1
+      , Index.insert newIndex FileInfo.empty mp
+      , 0 )
+
+    errorMsg = unwords
+        [ "VolatileDB invariant violation:"
+        , "The impossible happened!"
+        , "A file was found with less than "
+        , show n
+        , " blocks, but there are no files in db!"]
```

Comment:
You don't need the space after "than " and before " blocks", `unwords` is taking care of them already.

"the" missing before "db!".

I still don't really understand the message: does it mean that a file was found (with less than x blocks?) while we didn't expect to find one?

Also, can you put it closer to where it is used?

Reply:
Eventually, I changed a bit the implementation and it is not needed any more.